### ISSUE TITLE
Distributed training example + tests

### ISSUE THREAD

**Issue by jaanli** [blei-lab/edward#137]
Distributed training is a crucial feature. We need to make sure that at least all the examples run in a distributed way in the tests (multiple CPUs / multiple GPUs).

**Comment by dustinvtran** [comment]
I agree. I haven't checked it but my guess is that you can just wrap everything inside a device context, whether it be CPU or GPU. (It is all just defining a computational graph in TensorFlow and running an optimizer.) Batch training will be a different story as it depends on the inference algorithm.
